src/citrine/resources/predictor.py

```diff
@@ -125,6 +126,17 @@ def get(self,
         entity = self.session.get_resource(path, version=self._api_version)
         return self.build(entity)
 
+    def get_featurized_training_data(
+            self,
+            uid: Union[UUID, str],
+            *,
+            version: Union[int, str] = MOST_RECENT_VER
+    ) -> List[HierarchicalDesignMaterial]:
+        version_path = self._construct_path(uid, version)
+        full_path = f"{version_path}/featurized-training-data"
+        payload = self.session.get_resource(full_path, version=self._api_version)
```

Comment:
Based on how the backend streams the response out, it is not trivial to chunk the incoming JSON per element in the stream (see more details [here](https://doc.akka.io/docs/akka-http/current/routing-dsl/source-streaming-support.html#customising-response-rendering-mode)).

Because of that, I am simply loading the full JSON payload into memory and then parsing into the design material objects. We'll see if this ends up causing issues on bigger sample sizes, and can adjust accordingly then